miner/src/main/java/fr/raksrinana/channelpointsminer/miner/prediction/bet/outcome/MostTrustedPicker.java

```diff
@@ -0,0 +1,86 @@
+package fr.raksrinana.channelpointsminer.miner.prediction.bet.outcome;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import fr.raksrinana.channelpointsminer.miner.api.ws.data.message.subtype.Outcome;
+import fr.raksrinana.channelpointsminer.miner.database.IDatabase;
+import fr.raksrinana.channelpointsminer.miner.database.NoOpDatabase;
+import fr.raksrinana.channelpointsminer.miner.database.model.prediction.OutcomeStatistic;
+import fr.raksrinana.channelpointsminer.miner.handler.data.BettingPrediction;
+import fr.raksrinana.channelpointsminer.miner.prediction.bet.exception.BetPlacementException;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.extern.log4j.Log4j2;
+import org.jetbrains.annotations.NotNull;
+import java.util.Comparator;
+
+@JsonTypeName("mostTrusted")
+@Getter
+@EqualsAndHashCode
+@ToString
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Log4j2
+public class MostTrustedPicker implements IOutcomePicker{
+	
+	@Builder.Default
+	private int minTotalBetsPlacedByUser = 5;
+	@Builder.Default
+	private int minTotalBetsPlacedOnPrediction = 10;
+	@Builder.Default
+	private int minTotalBetsPlacedOnOutcome = 5;
+	
+	@Override
+	@NotNull
+	public Outcome chooseOutcome(@NotNull BettingPrediction bettingPrediction, @NotNull IDatabase database) throws BetPlacementException{
+		
+		try{
+			if(database instanceof NoOpDatabase){
+				throw new BetPlacementException("A database needs to be configured for this outcome picker to work");
+			}
+			
+			var outcomes = bettingPrediction.getEvent().getOutcomes();
+			var title = bettingPrediction.getEvent().getTitle();
+			
+			var outcomeStatistics = database.getOutcomeStatisticsForChannel(bettingPrediction.getEvent().getChannelId(), minTotalBetsPlacedByUser);
+			
+			var mostTrusted = outcomeStatistics.stream()
+					.max(Comparator.comparingDouble(OutcomeStatistic::getAverageReturnOnInvestment))
```

Comment:
I think it could be interesting to have the comparator configurable. Like there could be a field `StatisticComparator comparator` that the user can set through the config and would be passed to the comparator.

```java
@RequiredArgsConstructor
@Getter
enum StatisticComparator {
    ROI(Comparator.comparingDouble(OutcomeStatistic::getAverageReturnOnInvestment)),
    WIN(Comparator.comparingDouble(OutcomeStatistic::getWinRate));

    private final Comparator<OutcomeStatistic> comparator;
}
```

Reply:
I started out with the idea to compare the win rate. The reason why I dropped it is that what matters isn't the win rate, but how many points were earned. For example, if a user has the habit to bet on an outcome that has a 90% win chance but only a return ratio of 1.05. It's mathematically better to go with the user who bets on the outcome with 10% win chance and a return ratio of 20.

Reply:
Still having the choice is a good option I think. Can be set to ROI by default.

Maybe I'm wrong but it feels like it could be abused somehow. Especially because a bet itself is not considered worthy or not. By that I mean that if someone have a ROI of 10 because they bet 10 points or because they bet 50k is the same.
You can imagine that I'm an annoying user and will only bet on predictions that'll have a massive ROI (let's say 15), and bet only 10 points each time. I'm not risking much by doing that and overtime it could give me a somewhat nice ROI (if I win at least once every 15 bets, my ROI will slowly increase). However the win rate will be going towards 0. If the bot follows that user with predictions with higher amounts, like maybe 2% of your total amount, you may quickly drain your wallet as you'll lose a lot, then win big but on a lower base amount and repeat.

This scenario (having a user that is doing weird stuff) is quite unlikely I admit it, but I still prefer to give the choice to the user to find what fits their needs the best. Every channel is different and the people doing predictions too.

Reply:
Sure, giving a choice is fine. Especially if the predictions are mostly 50/50.

I think I understand your concern about predictions with a huge difference in return ratio between outcomes. Although I don't think these are common enough to bother. And wouldn't it be best, in that case, to not bet at all? If you bet on the prediction with the higher return-ratio but low win-chance you run the risk of bankrupting yourself fast (but with the chance to win big), if you bet on the one with the lower return-ratio and higher win-chance you slowly but surely drain your points.

Reply:
Maybe we can make a configurable limit to the return ratio. E.g. Don't bet on outcomes that have a return ratio of > 10, even if trustworthy users bet on it.